apps/studio/src/utils/sitemap.ts

```diff
@@ -75,11 +75,16 @@ const getSitemapTreeFromArray = (
     const summaryOfPage = indexPage?.summary
 
     return {
-      // NOTE: this is done for compat reasons as
-      // our publishing script uses the index page id
-      // if it can find one and if it cannot,
-      // then uses the base `resourceId`
-      id: indexPage?.id ?? resource.id,
+      id:
+        resource.type === ResourceType.Collection
+          ? // Note: Current publishing script still use collection id
+            // as the id even if there is an index page
+            resource.id
+          : // NOTE: this is done for compat reasons as
+            // our publishing script uses the index page id
+            // if it can find one and if it cannot,
+            // then uses the base `resourceId`
+            (indexPage?.id ?? resource.id),
```

Comment:
question: sorry missing a bit of context here, shouldn't we always use the Folder/Collection resource ID instead of the index page? Why do we need to use the index page's ID?

From what I understand that's what publishing script does, because the Folder/Collection resource is technically guaranteed to exist but not the index page (only an app-level guarantee).

Reply:
oh actually im  also not 100% sure - i realized this was changed but looking back i also don't 100% rmb/understand why it was changed. I only know the changes seem to be more about Folder rather than Collection, thus i made this change - was thinking of waiting for @seaerchin to be back but at least this PR can unblock the preview issue for agencies